storage-client/src/client.rs

```diff
@@ -36,11 +45,147 @@ impl StorageClientImpl {
+            let builder = ParquetRecordBatchStreamBuilder::new(file)
+                .await
+                .unwrap()
+                .with_batch_size(3);
+
+            let mask = ProjectionMask::all();
+            let stream = builder.with_projection(mask).build().unwrap();
+            let results = stream.try_collect::<Vec<_>>().await.unwrap();
+
+            // Return the record batch as a stream.
+            let (tx, rx) = channel(1);
+            tokio::spawn(async move {
+                for result in results {
+                    tx.send(result).await.unwrap();
+                }
+            });
+            Ok(rx)
+        } else {
+            Err(anyhow::anyhow!(
+                "Failed to download file. Response: {:?}",
+                response.status()
+            ))
+        }
     }
 
+    // TODO (kunle): I don't think this function is necessary.
     async fn request_data_sync(&self, _request: StorageRequest) -> Result<Vec<RecordBatch>> {
         todo!()
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use arrow::array::StringArray;
+    use mockito::Server;
+
+    /// WARNING: Put userdata1.parquet in the storage-node/tests/parquet directory before running this test.
+    #[tokio::test]
+    async fn test_storage_client_wo_ee_catalog() -> Result<()> {
+        let mut server = Server::new_async().await;
+        println!("server host: {}", server.host_with_port());
+        server
+            .mock("GET", "/file/userdata1.parquet")
+            .with_body_from_file("../storage-node/tests/parquet/userdata1.parquet")
+            .create_async()
+            .await;
+
+        let server_endpoint = server.url() + "/";
+        // The catalog endpoint is not used anyway. So randomly pass a url.
+        let storage_client = StorageClientImpl::new(&server_endpoint, "localhost:3031")?;
+        let request = StorageRequest::Table(1);
+        let mut receiver = storage_client.request_data(request).await?;
+        let mut record_batches = vec![];
+        while let Some(record_batch) = receiver.recv().await {
+            record_batches.push(record_batch);
+        }
+        assert!(!record_batches.is_empty());
+
+        let first_batch = &record_batches[0];
+        assert_eq!(first_batch.num_columns(), 13);
+
+        let first_names = StringArray::from(vec!["Amanda", "Albert", "Evelyn"]);
+        let last_names = StringArray::from(vec!["Jordan", "Freeman", "Morgan"]);
+        assert_eq!(
+            first_batch.column(2).as_any().downcast_ref::<StringArray>(),
+            Some(&first_names)
+        );
+        assert_eq!(
+            first_batch.column(3).as_any().downcast_ref::<StringArray>(),
+            Some(&last_names)
+        );
+
+        Ok(())
+    }
+
+    // /// WARNING: This test assumes that the data returned from the server is userdata1.parquet.
+    // /// This test validates the correctness of both the client and the server.
+    // #[tokio::test]
```

Comment:
We can add a `#[ignore = "<reason>"]` instead of commenting out all the test code

Reply:
Oops, we can actually delete it for now. I originally planned to move it to an e2e test. Will update it later.